### PR TITLE
#2299 Tests for pulling with versions

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Ref: #2299

What's done:
- Minor refactoring of tests for `ProbeMojo` and `PullMojo`
- Added disabled tests for `PullMojo` for testing versioning

---

## PR-Codex overview
This PR focuses on removing the `ChCompound` class and adding a new method `withVersionedHelloWorld()` to the `FakeMaven` class. 

### Detailed summary
- Removed the `ChCompound` class as it is no longer needed and has a doubtful design.
- Added the `withVersionedHelloWorld()` method to the `FakeMaven` class to add a correct versioned 'Hello world' program to the workspace.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please review this one

File: eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java [hunk 1]
```diff
@@ -154,22 +162,92 @@ void pullsUsingOfflineHash(@TempDir final Path temp) throws IOException {
...
+            PullMojoTest.exists(temp, PullMojoTest.VERSIONED),
+            Matchers.is(true)
+        );
+    }
+
+    @Disabled
+    @Test
+    void pullsProbedVersionedObjectFromOneObjectionary(@TempDir final Path temp)
+        throws IOException {
+        final CommitHash hash = new ChCached(new CommitHashesMap.Fake().get("0.28.5"));
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:194]: @maxonfjvipon Can we inline that variable?

File: eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java [hunk 2]
```diff
@@ -90,17 +98,17 @@ void pullsFromProbes(@TempDir final Path temp) throws IOException {
...
-            new Home(temp.resolve("target")).exists(
-                Paths.get(
-                    String.format(
-                        "%s/org/eolang/io/stdout.eo",
-                        PullMojo.DIR
+            .with(
+                "objectionaries",
+                new Objectionaries.Fake(
+                    new OyRemote(
+                        new ChCompound(null, null, "master")
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:105]: @maxonfjvipon Passing null values might be rather dangerous:
https://www.yegor256.com/2014/05/13/why-null-is-bad.html

File: eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java [hunk 3]
```diff
@@ -154,22 +162,92 @@ void pullsUsingOfflineHash(@TempDir final Path temp) throws IOException {
...
+                    new MapEntry<>(first, new OyRemote(first)),
+                    new MapEntry<>(second, new OyRemote(second)),
+                    new MapEntry<>(third, new OyRemote(third))
+                )
+            )
+            .with("withVersions", true)
+            .with("hsh", third)
+            .withVersionedProgram()
+            .execute(new FakeMaven.Pull());
+        MatcherAssert.assertThat(
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:227]: @maxonfjvipon Could you add assertion messages for all new tests, please?
https://www.yegor256.com/2023/01/19/layout-of-tests.html#assertions

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java (hunk 2 above)
Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:105]: @volodya-lombrozo I didn't touch constructing of this hash, just moved it

Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:105]: @volodya-lombrozo and implementation of the class assumes that some of the arguments can be `null`
```java
final CommitHash ret;
if (file == null && hash == null) {
    ret = new ChRemote(tag);
} else if (hash == null) {
     ret = new ChText(file, tag);
} else {
     ret = new ChPattern(hash, tag);
}
return new ChCached(ret);
```

Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:105]: @maxonfjvipon Looks like we have to remove that class :) What do you think?

Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/java/org/eolang/maven/PullMojoTest.java:105]: @volodya-lombrozo maybe, as I understand it's used only for the test purposes. I can make a todo to think about its fate later

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please have a look one more time

**Review by volodya-lombrozo — APPROVED** [review]
@maxonfjvipon Looks good to me, thanks!

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**